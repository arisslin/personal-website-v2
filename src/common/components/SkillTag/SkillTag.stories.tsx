import * as React from 'react';
import { Meta, Story } from '@storybook/react';
import SkillTagComponent, { SkillTagProps } from './SkillTag';

export default {
  title: 'Components/SkillTag',
  component: SkillTagComponent,
  argTypes: {
    text: {
      description: 'Text which get displayed inside the tag.',
    },
  },
  parameters: {
    docs: {
      description: {
        component: getDescription(),
      },
    },
  },
} as Meta;

function getDescription(): string {
  return (
    '<h2>About</h2>' +
    '<p>' +
    'The <b>SkillTag</b> was developed in Juli 2022. It displays the ' +
    'name of a propped skill.' +
    '</p>'
  );
}

export const SkillTag: Story<SkillTagProps> = (args) => (
  <SkillTagComponent text={args.text} />
);
SkillTag.args = {
  text: 'TypeScript',
};
