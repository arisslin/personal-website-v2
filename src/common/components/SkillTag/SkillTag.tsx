import * as React from 'react';
import styled from 'styled-components/macro';

type SkillTagProps = {
  text: string;
};

const SkillTag = ({ text }: SkillTagProps): JSX.Element => {
  if (!text) {
    return <></>;
  }

  return <SkillTagStyled data-testid={testId}>{text}</SkillTagStyled>;
};

const SkillTagStyled = styled.span`
  border-radius: 3px;
  display: inline-block;
  padding: 10px 12px;
  background-color: var(--color-yo);
  color: var(--color-light);
`;

const testId = 'skill-tag';

export default SkillTag;
export { SkillTagProps, testId };
