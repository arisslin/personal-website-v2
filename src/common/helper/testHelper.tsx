import { cleanup, fireEvent, render } from '@testing-library/react';

/**
 * Does a snapshot test for component it got.
 * @param {JSX.Element} component to creating snapshot for
 * @param {string} variant to create a postfix for the displayed test string
 * @return {void}
 */
export function doSnapshotTest(component: JSX.Element, variant?: string): void {
  const postfix = `${component.type.name as string} ${variant ? variant : ''}`;

  test(`Snapshot test for ${postfix}`, () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
    cleanup();
  });
}

/**
 * Fires an click event with 'fireEvent' from '@testing-library/react' on an
 * element.
 * @param {HTMLElement} element to fire click on.
 * @param {boolean} bubbles attribute of 'MouseEvent'.
 * @param {boolean} cancelable attribute of 'MouseEvent'.
 * @return {void}
 */
export function fireClickEvent(
  element: Element,
  bubbles: boolean,
  cancelable: boolean
) {
  fireEvent(element, new MouseEvent('click', { bubbles, cancelable }));
}
