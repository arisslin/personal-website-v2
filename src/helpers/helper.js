export const getText = (wantText) => {
  if (typeof wantText === 'boolean') {
    return wantText ? 'text' : null;
  }

  return null;
};
