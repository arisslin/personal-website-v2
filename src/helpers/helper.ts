export const getText = (wantText: boolean) => (wantText ? 'text' : null);
