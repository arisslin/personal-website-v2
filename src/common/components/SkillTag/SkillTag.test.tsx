import * as React from 'react';
import { render, screen } from '@testing-library/react';
import SkillTag, { testId } from './SkillTag';
import { doSnapshotTest } from '../../helper/testHelper';

describe('SkillTag', () => {
  const text = 'TypeScript';
  const component = <SkillTag text={text} />;

  it('renders with text', () => {
    render(component);

    expect(screen.getByText(text)).toBeInTheDocument();
  });

  it('does not render if text is an empty string', () => {
    render(<SkillTag text='' />);

    expect(screen.queryByTestId(testId)).not.toBeInTheDocument();
  });

  doSnapshotTest(component);
});
