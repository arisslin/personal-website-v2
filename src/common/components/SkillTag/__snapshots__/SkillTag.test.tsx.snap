// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillTag Snapshot test for SkillTag  1`] = `
<div>
  <span
    class="SkillTag__SkillTagStyled-sc-pqp8qz-0 eUeqiS"
    data-testid="skill-tag"
  >
    TypeScript
  </span>
</div>
`;
